cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -327,6 +327,11 @@ func (scaleSet *ScaleSet) TargetSize() (int, error) {
 
 // IncreaseSize increases Scale Set size
 func (scaleSet *ScaleSet) IncreaseSize(delta int) error {
+	if scaleSet.curSize == -1 {
+		klog.V(1).Infof("the scale set %s is under initialization, would skip the operation", scaleSet.Name)
+		return nil
+	}
+
```

Comment:
I'd also move this check under the line where it gets the currSize.

The 
```
	size, err := scaleSet.GetScaleSetSize()
	if err != nil {
		return err
	}
```
happens under a mutex so it should be safer.